daemon/config.go

```diff
@@ -74,3 +111,102 @@ func (config *Config) InstallCommonFlags(cmd *flag.FlagSet, usageFn func(string)
 	cmd.StringVar(&config.ClusterStore, []string{"-cluster-store"}, "", usageFn("Set the cluster store"))
 	cmd.Var(opts.NewMapOpts(config.ClusterOpts, nil), []string{"-cluster-store-opt"}, usageFn("Set cluster store options"))
 }
+
+func parseClusterAdvertiseSettings(clusterStore, clusterAdvertise string) (string, error) {
+	if clusterAdvertise == "" {
+		return "", errDiscoveryDisabled
+	}
+	if clusterStore == "" {
+		return "", fmt.Errorf("invalid cluster configuration. --cluster-advertise must be accompanied by --cluster-store configuration")
+	}
+
+	advertise, err := discovery.ParseAdvertise(clusterAdvertise)
+	if err != nil {
+		return "", fmt.Errorf("discovery advertise parsing failed (%v)", err)
+	}
+	return advertise, nil
+}
+
+// ReloadConfiguration reads the configuration in the host and reloads the daemon and server.
+func ReloadConfiguration(configFile string, flags *flag.FlagSet, reload func(*Config)) {
+	logrus.Infof("Reloading daemon configuration, %s modified", configFile)
+	newConfig, err := readConfigFromFile(configFile, flags)
+	if err != nil {
+		logrus.Error(err)
+	} else {
+		reload(newConfig)
+	}
+}
+
+// MergeDaemonConfigurations reads a configuration file and uses its configuration
+// as the base to apply the given configuration.
+// It fails with a conflict if there are flags set that are also present in the configuration file.
+func MergeDaemonConfigurations(config *Config, flags *flag.FlagSet, configFile string) (*Config, error) {
+	c, err := readConfigFromFile(configFile, flags)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := mergo.Merge(c, config); err != nil {
```

Comment:
Probably related to the function description, but would be great to comment here explaining that `c` is the destination and `config` the source.